simple_mediation.rmd

```diff
@@ -0,0 +1,123 @@
+```{r, include=FALSE}
+options(tinytex.verbose = TRUE)
+```
+
+```{r}
+#RQ: Does domain-specifical movement behaviour profile membership mediate the relationship between socioeconomic position and socio-emotional outcomes in children?
+##Simple mediation for males
+```
+
+```{r include=FALSE}
+library(medflex)
+library(dplyr)
+library(haven)
+library(rmarkdown)
+```
+
+```{r}
+# Load male dataset from Study 2, LSAC wave 6 data, and LSAC wave 8 data
+load(file = "Z:/LSAC dataset/Study_2/Study_2/df_males_domsp_weekday.RData")
+lsac_wave6 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb10.sas7bdat")
+lsac_wave8 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb14.sas7bdat")
+```
+
+```{r}
+# Select relevant columns from male dataset (hicid, profile membership, age, remoteness, Indigenous status, SEP)
+relevant_male_data <- df_male_domsp_weekday %>%
+  select(hicid, model3_trajectory_assignments, Age, Remoteness, Indigenous, SEP)
+
+# Select relevant columns from LSAC wave 6 (SEIFA varialbes), parents living at home, parent mental health, parent ethnicity)
+wave_6_relevant_data <- lsac_wave6 %>%
+  select(hicid, fcnfsad2d, fcnfsda2d, fcnfser2d, fcnfseo2d, fmoth, ffath, fsc13a1i, zf09fm, zf09ff) 
+
+# Update column names to make sense
+colnames(wave_6_relevant_data)[2:10] <- c("adv", "dis", "resources", "education_occupation", "mother_home", "father_home", "mental_health", "mother_race", "father_race")
+
+# Select relevant columns from LSAC Wave 8 (SDQ results)
+wave_8_relevant_data <- lsac_wave8 %>%
+  select(hicid, hapsoc:hasdqtb) 
+
+# Update column names to make sense
+colnames(wave_8_relevant_data)[2:7] <- c("prosocial", "hyperactivity", "emotional", "peer", "conduct", "total")
+
+# Combine columns to make new dataset
+domsp_wave6 <- inner_join(relevant_male_data, wave_6_relevant_data, by = "hicid")
+df_males <- inner_join(domsp_wave6, wave_8_relevant_data, by = "hicid")
+``` 
+
+```{r}
+#Take out participants with missing covariate data 
+colSums(df_males < -8)
+# 2 mental health, 9 mother race, 125 father race
+df_males <- df_males[!(df_males$mental_health == -9), ]
+df_males <- df_males[!(df_males$mother_race == -9), ]
+df_males <- df_males[!(df_males$father_race == -9), ]
```

Comment:
Just checking that you looked at the codebook (and other docs) to see if -9 had any kind of meaning. E.g., could it just be that they don't have fathers (or mothers don't know the fathers race)? If so, you could code these as a new variable rather than just removing. 125 is ~12% of your sample, so worth checking for other options before removing.